cpp/src/streaming/cudf/parquet.cpp

```diff
@@ -90,9 +91,21 @@ Node produce_chunks(
         co_await ctx->executor()->schedule();
         // TODO: This reads the metadata ntasks times.
         // See https://github.com/rapidsai/cudf/issues/20311
-        auto sent = co_await ticket->send(
-            read_parquet_chunk(ctx, stream, chunk_options, chunk.sequence_number)
-        );
+        auto [msg, exception] = [&]() -> std::pair<Message, std::exception_ptr> {
```

Comment:
Would you recommend using this pattern in all of our utilities calling cudf operations?

Reply:
No, this is because we can't use the `ShutdownAtExit` utility because `produce_chunks` is used in a multi-producer setup. Usually we use `ShutdownAtExit` to shutdown channels if an exception is not caught. But it _also_ shuts down the channel when the coroutine finishes.

But if we have two `produce_chunks` into the same output channel, we don't want to shut down the output channel if one `produce_chunks` is finished, only if an exception occurred.